internal/image_template.go

```diff
@@ -33,8 +34,25 @@ func (t *ImageTemplate) String() string {
 	return t.Raw
 }
 
+func prepTemplateString(input string) string {
+	output := ""
+	matches := TemplateRegex.FindAllStringSubmatchIndex(input, -1)
+	start := 0
+	for _, match := range matches {
+		output += input[start:match[2]] + "index ."
+		parts := strings.Split(input[match[2]+1:match[3]], ".")
+		for _, part := range parts {
+			output += " \"" + part + "\""
+		}
+		start = match[3]
+	}
+	output += input[start:]
+	return output
+}
+
 func NewFromString(input string) (*ImageTemplate, error) {
-	temp, err := template.New(input).Parse(input)
+	preppedInput := prepTemplateString(input)
+	temp, err := template.New(preppedInput).Parse(preppedInput)
```

Comment:
This is probably nitpicking, but curious since I am not that familiar with templating. The input of template.New does not need to be the input, but just an identifier right? Identifier that could reuse? Do you know what's the purpose of that identifier, I can not find it in the template package code.